edalize/symbiflow.py

```diff
@@ -0,0 +1,176 @@
+                        {'name' : 'builddir',
+                         'type' : 'String',
+                         'desc' : 'directory where all the intermediate files will be stored (default "build")'},
+                        {'name' : 'vendor',
+                         'type' : 'String',
+                         'desc' : 'Target architecture. Currently only "xilinx" is supported'},
+                        {'name' : 'pnr',
+                         'type' : 'String',
+                         'desc' : 'Place and Route tool. Currently only "vpr" and "nextpnr" are supported'},
+                   ]}
+
+            symbiflow_members = symbiflow_help['members']
+
+            return {'description' : "The Symbilow backend executes Yosys sythesis tool and VPR place and route. It can target multiple different FPGA vendors",
+                    'members': symbiflow_members}
+
+    def get_version(self):
+        return "1.0"
+
+    def configure_nextpnr(self):
+        (src_files, incdirs) = self._get_fileset_files(force_slash=True)
+
+        nextpnr_edam = {
+                'files'         : self.files,
+                'name'          : self.name,
+                'toplevel'      : self.toplevel,
+                'tool_options'  : {'nextpnr' : {
+                                        'arch' : 'xilinx',
+                                        'nextpnr_as_subtool' : True,
+                                        }
+
+                                }
+                }
+
+        nextpnr = getattr(import_module("edalize.nextpnr"), 'Nextpnr')(nextpnr_edam, self.work_root)
+        nextpnr.configure()
+
+        builddir = self.tool_options.get('builddir', 'build')
+
+        part = self.tool_options.get('part', None)
+        package = self.tool_options.get('package', None)
+
+        assert part is not None, 'Missing required "part" parameter'
+        assert package is not None, 'Missing required "package" parameter'
+
+        partname = part + package
+
+        if 'xc7a' in part:
+            bitstream_device = 'artix7'
+        if 'xc7z' in part:
+            bitstream_device = 'zynq7'
+        if 'xc7k' in part:
+            bitstream_device = 'kintex7'
+
+        makefile_params = {
+                'top' : self.name,
+                'partname' : partname,
+                'bitstream_device' : bitstream_device,
+                'builddir' : builddir,
+            }
+
+        self.render_template('symbiflow-nextpnr-makefile.j2',
+                             'Makefile',
+                             makefile_params)
+
+    def configure_vpr(self):
+        (src_files, incdirs) = self._get_fileset_files(force_slash=True)
+
+        has_vhdl     = 'vhdlSource'      in [x.file_type for x in src_files]
+        has_vhdl2008 = 'vhdlSource-2008' in [x.file_type for x in src_files]
+
+        assert (not has_vhdl and not has_vhdl2008), 'VHDL files are not supported in Yosys'
```

Comment:
I don't think you need to check the files twice, and an assertion is probably not correct for incorrect user input. Perhaps the following?

```suggestion
        has_vhdl = [x.name for x in src_files if x.file_type.startsWith("vhdlSource")

        if has_vhdl:
            logger.error("VHDL files are not supported in Yosys: {}".format(", ".join(has_vhdl)))
```

Reply:
Done